src/sm/sm-nist.c

```diff
@@ -1857,13 +1868,36 @@ sm_nist_post_transmit(sc_card_t *card, const struct iso_sm_ctx *ctx,
 	SC_FUNC_RETURN(card->ctx, SC_LOG_DEBUG_SM, r);
 }
 
-//static int
-//nist_sm_close(sc_card_t card)
-//{
-//	sm_nist_private_data_t *priv;
-//	 if (card) {
-//	 }
-//}
+
+static int
+sm_nist_close(sc_card_t *card)
+{
+	struct iso_sm_ctx *sm_ctx;
+	sm_nist_private_data_t *priv;
+
+	if (!card)
+		return SC_ERROR_INVALID_ARGUMENTS;
+
+	SC_FUNC_CALLED(card->ctx, SC_LOG_DEBUG_SM);
+
+	if (((sm_ctx = ISO_CTX_FROM_CARD)) && ((priv = SM_NIST_PRIV(sm_ctx)))) {
+		/*
+		 * a SM failure while using SM 00 20 00 xx lc=0
+		 * from reader_lock_obtained maybe caused
+		 * by interference of other process that started its own 
+		 * SM session. In this case do not close and let reader_lock_obtained
+		 * continue.
```

Comment:
Could you elaborate on what happened here? I would not expect multiple processes to use *different* SM keys with *the same* OpenSC context (i.e. sm_ctx).

Reply:
Each process sets up its own keys, an ephemeral EC key is used as part of the SM initialization,  and the application and card each have counters or number of transactions.  Each process has has its own sm_ctx. So if process tries to use SM it will cause an error for the other process. 

The comment "In this case do not close and let reader_lock_obtained  continue" is the way the SM is telling the "reader_lock_obtain" to reopen the SM before continuing.